airflow/providers/google/cloud/operators/vertex_ai/custom_job.py

```diff
@@ -180,6 +180,38 @@ def execute(self, context: Context) -> None:
             stacklevel=2,
         )
 
+    def execute_complete(self, context: Context, event: dict[str, Any]) -> dict[str, Any] | None:
```

Comment:
@e-galan I have noticed that this method returns a Job object as an operator's result, but in `sync` mode the operator returns a `Model` object or None as a result. Could we make a refactor for this method and return for `deff` mode the same result as for `sync` mode?

Reply:
It happens because we have to use a different approach with the deferrable mode. 

In the sync mode we submit the training pipeline with `sync=True` and then just wait for the method to finish and return the model instance.

In the case of the def mode, we submit the training pipeline and then have to exit. When we exit the `submit()`, our job ID is still the ID of the training pipeline, because the custom job ID has not been generated yet. 

Since the training pipeline ID is the only thing we can use in the def mode, we check the pipeline status instead of custom job status, and once the training pipeline is finished, we return it in a serialized format.

Reply:
I think we want to get a model object as the result, then we'll have to instantiate another client ([ModelServiceClient](https://cloud.google.com/python/docs/reference/aiplatform/1.19.0/google.cloud.aiplatform_v1.services.model_service.ModelServiceClient)) and use its methods to get that. 

We'll have to do it inside the execute_complete() method.

Reply:
Fixed by extracting the _model_to_upload_ key data from the def mode response.